R/PseudoBulk.R

```diff
@@ -360,48 +360,65 @@ CreateStudyWideBarPlot <- function(pairwise_de_results, pvalue_threshold = 0.05,
 #' @param design a design/model matrix returned by DesignModelMatrix().
-#' @param contrast_columns The vector of contrast names supplied to DesignModelMatrix() upstream of this function. Please ensure this vector is identical to the contrast_columns vector supplied to DesignModelMatrix().
 #' @param use_require_identical_logic Whether or not to apply require_identical logic using require_identical_fields. It is possible to use logic gates to keep metadata fields constant, but using this saves time & effort/iteration.
 #' @param require_identical_fields The metadata columns of the SeuratObj that you wish to keep constant. For example: defining "Tissue" in this vector would filter all contrasts comparing Liver to PBMC, as their tissue is not identical.
 #' @param filtered_contrasts_output_file The file to write the list of filtered contrasts to (one pair of samples per row, separated into their metadata fields).
 #' @return A dataframe of pairwise contrasts, with column names indicating the directionality (positive vs negative) and associated metadata column (inferred by contrast_columns).
 #' @examples
 #' \dontrun{
-#' #set up design matrix
-#' contrast_columns = c("Timepoint", "Vaccine", "Challenge", "Tissue", "cell_type", "SampleType")
-#' design <- DesignModelMatrix(seuratObj, contrast_columns = c("Timepoint", "Vaccine", "Challenge", "Tissue", "cell_type","SampleType"), sampleIdCol = 'cDNA_ID')
+#' #Set up design matrix
+#' design <- DesignModelMatrix(pseudobulked_seuratObj, 
+#'                             contrast_columns = c('Timepoint', 'Vaccine', 'Challenge', 'Tissue', 'cell_type', 'SampleType'), 
+#'                             sampleIdCol = 'cDNA_ID')
 #' 
 #' 
-#' #form study design using logic gates. Each gate consists of a metadata field (field_to_check) to apply the logic gate to 
-#' gate_one <- data.frame(field_to_check = "Challenge", field_logic = "xor", criteria = "Mock.challenged")
-#' gate_two <- data.frame(field_to_check = "SampleType", field_logic = "all", criteria = "Necropsy")
-#' logical_dataframe <- rbind(gate_one, gate_two)
-#' require_identical_fields <- c("Tissue","cell_type", "SampleType")
+#' #Form the study design. This should be a list of gates (each specified as a list itself) with three elements. 
+#' #The first element defines the metadata column or field that will be used for filtering. 
+#' #The second defines the logic gate (AND -> any, OR -> any, xor, xnor, nand, nor). 
+#' #The third defines the specific value of the metadata field that will be tested for equivalence against the positive and negative contrasts within the gate.
+#' logic_list <- list(
```

Comment:
You can decide if worth it, but given the complexity and given we want this vignette to be useful to other people in the lab, is there a reasonable way to mock of a simple table, and illustrate some examples of what rows different logic gates would or would not return? It would become long in the code, but that's not that big a deal. I think you could make minimal table with a few columns and rows, and then maybe show a couple of gates one at a time, and show which rows they would select?